custom_components/powerocean/ecoflow.py

```diff
@@ -65,189 +86,234 @@ def authorize(self):
 
-        if response_message.lower() != "success":
-            raise Exception(f"{response_message}")
+        # Validate message key
+        message = response_data.get("message")
+        if message is None:
+            raise KeyError(f"'message' key missing in response: {response_data}")
 
-        return response
+        if message.lower() != "success":
+            raise Exception(f"API Error: {message}")
+
+        return response_data
 
     # Fetch the data from the PowerOcean device, which then constitues the Sensors
-    def fetch_data(self):
+    def fetch_data(self) -> dict:
         """Fetch data from Url."""
         url = self.url_user_fetch
         try:
             headers = {
                 "authorization": f"Bearer {self.token}",
-                # "accept": "application/json, text/plain, */*",
                 "user-agent": "Firefox/133.0",
-                # "platform": "web",
-                "product-type": "83",
+                "product-type": self.ecoflow_variant,
             }
             request = requests.get(self.url_user_fetch, headers=headers, timeout=30)
             response = self.get_json_response(request)
 
-            try:
-                # TESTING!!! create response file and use it as data
-                with Path.open(mocked_response, "r", encoding="utf-8") as datei:
-                    response = json_loads(datei.read())
-            except FileExistsError:
-                error = f"Data from: {mocked_response}"
-                _LOGGER.warning(error)
-            except FileNotFoundError:
-                error = f"Responsefile not present: {mocked_response}"
-                _LOGGER.debug(error)
+            if self.use_mocked_response:
+                try:
+                    with Path.open(mocked_response, "r", encoding="utf-8") as file:
+                        response = json_loads(file.read())
+                except FileNotFoundError:
+                    _LOGGER.debug(
+                        f"Mocked response file not present: {mocked_response}"
+                    )
 
             _LOGGER.debug(f"{response}")
-            return self._get_sensors(response)
+            # Ensure response is a dictionary before passing to _get_sensors
+            if isinstance(response, dict):
+                return self._get_sensors(response)
+            raise ResponseTypeError(type(response).__name__)
 
-        except ConnectionError:
-            error = f"ConnectionError in fetch_data: Unable to connect to {url}. Device might be offline."
+        except ConnectionError as err:
+            error = f"ConnectionError in fetch_data: Unable to connect to {url}."
             _LOGGER.warning(error + ISSUE_URL_ERROR_MESSAGE)
-            raise IntegrationError(error)
+            raise IntegrationError(error) from err
+
+    def __get_unit(self, key: str) -> str | None:
+        """Get unit from key name using a dictionary mapping."""
+        unit_mapping = {
```

Comment:
(First of all, these are great improvements, thanks!)

Minor nit on this: Even cleaner would be to only have lowercase and normalize when you do the lookup. Then you handle any format that can come in the future.